c/src/atom.rs

```diff
@@ -1006,42 +1010,81 @@ impl bindings_set_t {
     }
 }
 
-pub type bindings_mut_callback_t = lambda_t<*mut bindings_t>;
+/// @brief Function signature for a callback providing mutable access to individual Bindings frames within a Bindings Set
+/// @ingroup matching_group
+/// @param[in]  bindings  A pointer to the `bindings_t`.  It is ok to call functions that modify the `bindings_t` within the callback
+/// @param[in]  context  The context state pointer initially passed to the upstream function initiating the callback.
+/// @warning The ability for the function to mutate the `bindings_t` may be deprecated in the future, so please use this ability cautiously.
+///
+//TODO: QUESTION FOR VITALY: Is there a good reason to keep this mutable?  I think we would be better off with an immutable accessor, as we update the APIs for the non-determinism
```

Comment:
I believe I made it mutable because in my version of the API it was required when called function gets ownership over passed argument. Thus `* mut` can be replaced by moving the passed instance.

Reply:
Yes, I think it should work for this case.